<Type Name="NullReferenceException" FullName="System.NullReferenceException">
  <Metadata>
    <Meta Name="ms.openlocfilehash" Value="871adebd50877699047b62f9d350cf2555dfe480" />
    <Meta Name="ms.sourcegitcommit" Value="d31dc2ede16f6f7bc64e90d9f897ff54c4e3869b" />
    <Meta Name="ms.translationtype" Value="HT" />
    <Meta Name="ms.contentlocale" Value="ja-JP" />
    <Meta Name="ms.lasthandoff" Value="04/03/2018" />
  </Metadata>
  <TypeSignature Language="C#" Value="public class NullReferenceException : SystemException" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi serializable beforefieldinit NullReferenceException extends System.SystemException" />
  <TypeSignature Language="DocId" Value="T:System.NullReferenceException" />
  <TypeSignature Language="VB.NET" Value="Public Class NullReferenceException&#xA;Inherits SystemException" />
  <TypeSignature Language="C++ CLI" Value="public ref class NullReferenceException : SystemException" />
  <AssemblyInfo>
    <AssemblyName>System.Runtime</AssemblyName>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
    <AssemblyVersion>4.0.10.0</AssemblyVersion>
    <AssemblyVersion>4.0.20.0</AssemblyVersion>
    <AssemblyVersion>4.1.0.0</AssemblyVersion>
    <AssemblyVersion>4.2.0.0</AssemblyVersion>
    <AssemblyVersion>4.2.1.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>mscorlib</AssemblyName>
    <AssemblyVersion>2.0.5.0</AssemblyVersion>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>netstandard</AssemblyName>
    <AssemblyVersion>2.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <Base>
    <BaseTypeName>System.Exception</BaseTypeName>
    <BaseTypeName FrameworkAlternate="netframework-4.5">System.SystemException</BaseTypeName>
  </Base>
  <Interfaces />
  <Attributes>
    <Attribute>
      <AttributeName>System.Runtime.InteropServices.ComVisible(true)</AttributeName>
    </Attribute>
  </Attributes>
  <Docs>
    <summary>
      <span data-ttu-id="72d9b-101">null オブジェクト参照を逆参照しようとした場合にスローされる例外。</span>
      <span class="sxs-lookup">
        <span data-stu-id="72d9b-101">The exception that is thrown when there is an attempt to dereference a null object reference.</span>
      </span>
    </summary>
    <remarks>
      <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="72d9b-102">A<xref:System.NullReferenceException>を値型のメンバーにアクセスしようとすると、例外がスローされます`null`です。</span><span class="sxs-lookup"><span data-stu-id="72d9b-102">A <xref:System.NullReferenceException> exception is thrown when you try to access a member on a type whose value is `null`.</span></span> <span data-ttu-id="72d9b-103">A<xref:System.NullReferenceException>例外は、通常開発者のエラーを反映し、次のシナリオでがスローされます。</span><span class="sxs-lookup"><span data-stu-id="72d9b-103">A <xref:System.NullReferenceException> exception typically reflects developer error and is thrown in the following scenarios:</span></span>  
  
-   <span data-ttu-id="72d9b-104">忘れた場合、参照型のインスタンスを作成します。</span><span class="sxs-lookup"><span data-stu-id="72d9b-104">You've forgotten to instantiate a reference type.</span></span> <span data-ttu-id="72d9b-105">次の例では、`names`が宣言されていますが、インスタンス化されることはありません。</span><span class="sxs-lookup"><span data-stu-id="72d9b-105">In the following example, `names` is declared but never instantiated:</span></span>  
  
     [!code-csharp[System.NullReferenceException.Class#1](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.nullreferenceexception.class/cs/example1.cs#1)]
     [!code-vb[System.NullReferenceException.Class#1](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.nullreferenceexception.class/vb/example1.vb#1)]  
  
     <span data-ttu-id="72d9b-106">このコードをコンパイルするときに、一部のコンパイラは警告を発行します。</span><span class="sxs-lookup"><span data-stu-id="72d9b-106">Some compilers issue a warning when they compile this code.</span></span> <span data-ttu-id="72d9b-107">他のユーザーが、エラーを発行し、コンパイルは失敗します。</span><span class="sxs-lookup"><span data-stu-id="72d9b-107">Others issue an error, and the compilation fails.</span></span> <span data-ttu-id="72d9b-108">この問題を解決するには、オブジェクトのインスタンス、その値が不要になったように`null`です。</span><span class="sxs-lookup"><span data-stu-id="72d9b-108">To address this problem, instantiate the object so that its value is no longer `null`.</span></span> <span data-ttu-id="72d9b-109">次の例ではこの型のクラスのコンス トラクターを呼び出すことによって行われます。</span><span class="sxs-lookup"><span data-stu-id="72d9b-109">The following example does this by calling a type's class constructor.</span></span>  
  
     [!code-csharp[System.NullReferenceException.Class#2](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.nullreferenceexception.class/cs/example1a.cs#2)]
     [!code-vb[System.NullReferenceException.Class#2](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.nullreferenceexception.class/vb/example1a.vb#2)]  
  
-   <span data-ttu-id="72d9b-110">これを初期化する前に配列の次元を忘れた場合。</span><span class="sxs-lookup"><span data-stu-id="72d9b-110">You've forgotten to dimension an array before initializing it.</span></span> <span data-ttu-id="72d9b-111">次の例では、`values`整数の配列として宣言されたが、それに含まれる要素の数が指定されていません。</span><span class="sxs-lookup"><span data-stu-id="72d9b-111">In the following example, `values` is declared to be an integer array, but the number of elements that it contains is never specified.</span></span> <span data-ttu-id="72d9b-112">スローされたため、その値を初期化しないように、<xref:System.NullReferenceException>例外。</span><span class="sxs-lookup"><span data-stu-id="72d9b-112">The attempt to initialize its values therefore thrown a  <xref:System.NullReferenceException> exception.</span></span>  
  
     [!code-csharp[System.NullReferenceException.Class#10](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.nullreferenceexception.class/cs/Array3.cs#10)]
     [!code-vb[System.NullReferenceException.Class#10](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.nullreferenceexception.class/vb/Array3.vb#10)]  
  
     <span data-ttu-id="72d9b-113">例外は、次の例のように、初期化する前に、配列内の要素の数を宣言することで回避できます。</span><span class="sxs-lookup"><span data-stu-id="72d9b-113">You can eliminate the exception by declaring the number of elements in the array before initializing it, as the following example does.</span></span>  
  
     [!code-csharp[System.NullReferenceException.Class#11](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.nullreferenceexception.class/cs/Array4.cs#11)]
     [!code-vb[System.NullReferenceException.Class#11](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.nullreferenceexception.class/vb/Array4.vb#11)]  
  
     <span data-ttu-id="72d9b-114">宣言して、配列の初期化の詳細については、次を参照してください。[配列](http://msdn.microsoft.com/library/bb79bdde-e570-4c30-adb0-1dd5759ae041)と[配列](http://msdn.microsoft.com/library/dbf29737-b589-4443-bee6-a27588d9c67e)です。</span><span class="sxs-lookup"><span data-stu-id="72d9b-114">For more information on declaring and initializing arrays, see [Arrays](http://msdn.microsoft.com/library/bb79bdde-e570-4c30-adb0-1dd5759ae041) and [Arrays](http://msdn.microsoft.com/library/dbf29737-b589-4443-bee6-a27588d9c67e).</span></span>  
  
-   <span data-ttu-id="72d9b-115">取得する、 **null** 、メソッドから値を返し、返される型のメソッドを呼び出します。</span><span class="sxs-lookup"><span data-stu-id="72d9b-115">You get a **null** return value from a method, and then call a method on the returned type.</span></span> <span data-ttu-id="72d9b-116">ドキュメント エラー; の結果は場合があります。メソッドの呼び出しが戻ることに注意してくださいにドキュメントが失敗した`null`です。</span><span class="sxs-lookup"><span data-stu-id="72d9b-116">This sometimes is the result of a documentation error; the documentation fails to note that a method call can return `null`.</span></span> <span data-ttu-id="72d9b-117">それ以外の場合に、コード誤って前提としています、メソッドが常に返される以外**null**値。</span><span class="sxs-lookup"><span data-stu-id="72d9b-117">In other cases, your code erroneously assumes that the method will always return a non-**null** value.</span></span>  
  
     <span data-ttu-id="72d9b-118">次の例のコードでは、<xref:System.Array.Find%2A?displayProperty=nameWithType>メソッドは常に返します`Person`オブジェクト`FirstName`フィールドに一致する検索文字列。</span><span class="sxs-lookup"><span data-stu-id="72d9b-118">The code in the following example assumes that the <xref:System.Array.Find%2A?displayProperty=nameWithType> method always returns `Person` object whose `FirstName` field matches a search string.</span></span> <span data-ttu-id="72d9b-119">一致が存在しないために、ランタイム、<xref:System.NullReferenceException>例外。</span><span class="sxs-lookup"><span data-stu-id="72d9b-119">Because there is no match, the runtime throws a <xref:System.NullReferenceException> exception.</span></span>  
  
     [!code-csharp[System.NullReferenceException.Class#4](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.nullreferenceexception.class/cs/nullreturn2.cs#4)]
     [!code-vb[System.NullReferenceException.Class#4](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.nullreferenceexception.class/vb/nullreturn2.vb#4)]  
  
     <span data-ttu-id="72d9b-120">この問題を解決するには、テストされていないことを確認するメソッドの戻り値`null`ように次の例では、そのメンバーを呼び出す前にします。</span><span class="sxs-lookup"><span data-stu-id="72d9b-120">To address this problem, test the method's return value to ensure that it is not `null` before calling any of its members, as the following example does.</span></span>  
  
     [!code-csharp[System.NullReferenceException.Class#5](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.nullreferenceexception.class/cs/nullreturn2a.cs#5)]
     [!code-vb[System.NullReferenceException.Class#5](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.nullreferenceexception.class/vb/nullreturn2a.vb#5)]  
  
-   <span data-ttu-id="72d9b-121">式を使用している (たとえばをチェーンしてメソッドまたはプロパティの一覧まとめて) 値を取得して、値は、かどうかをチェックする場合は`null`、ランタイムはまだをスロー、<xref:System.NullReferenceException>例外。</span><span class="sxs-lookup"><span data-stu-id="72d9b-121">You're using an expression (for example, you're chaining a list of methods or properties together) to retrieve a value and, although you're checking whether the value is `null`, the runtime still throws a <xref:System.NullReferenceException> exception.</span></span> <span data-ttu-id="72d9b-122">これが発生するため、式の中間値のいずれかの`null`します。</span><span class="sxs-lookup"><span data-stu-id="72d9b-122">This occurs because one of the intermediate values in the expression returns `null`.</span></span> <span data-ttu-id="72d9b-123">その結果、用、残りの部分`null`は評価されません。</span><span class="sxs-lookup"><span data-stu-id="72d9b-123">As a result, your rest for `null` is never evaluated.</span></span>  
  
     <span data-ttu-id="72d9b-124">次の例では定義、`Pages`によって提示される web ページに関する情報をキャッシュするオブジェクト`Page`オブジェクト。</span><span class="sxs-lookup"><span data-stu-id="72d9b-124">The following example defines a `Pages` object that  caches information about web pages, which are presented by `Page` objects.</span></span> <span data-ttu-id="72d9b-125">`Example.Main`メソッド セルの現在の web ページに null 以外のタイトルが付いて場合は、タイトルが表示されます。</span><span class="sxs-lookup"><span data-stu-id="72d9b-125">The `Example.Main` method checks whether the current web page has a non-null title and, if it does, displays the title.</span></span> <span data-ttu-id="72d9b-126">この確認に関係なく、メソッドをスロー、<xref:System.NullReferenceException>例外。</span><span class="sxs-lookup"><span data-stu-id="72d9b-126">Despite this check, however, the method throws a <xref:System.NullReferenceException> exception.</span></span>  
  
     [!code-csharp[System.NullReferenceException.Class#6](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.nullreferenceexception.class/cs/Chain1.cs#6)]
     [!code-vb[System.NullReferenceException.Class#6](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.nullreferenceexception.class/vb/Chain1.vb#6)]  
  
     <span data-ttu-id="72d9b-127">例外がスローされます`pages.CurrentPage`返します`null`キャッシュにページの情報が格納されていない場合。</span><span class="sxs-lookup"><span data-stu-id="72d9b-127">The exception is thrown because `pages.CurrentPage` returns `null` if no page information is  stored in the cache.</span></span> <span data-ttu-id="72d9b-128">値をテストすることによってこの例外を修正することができます、`CurrentPage`現在を取得する前にプロパティ`Page`オブジェクトの`Title`プロパティ、次の例としては。</span><span class="sxs-lookup"><span data-stu-id="72d9b-128">This exception can be corrected by testing the value of the `CurrentPage` property before retrieving the current `Page` object's `Title` property, as the following example does:</span></span>  
  
     [!code-csharp[System.NullReferenceException.Class#7](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.nullreferenceexception.class/cs/Chain2.cs#7)]
     [!code-vb[System.NullReferenceException.Class#7](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.nullreferenceexception.class/vb/Chain2.vb#7)]  
  
-   <span data-ttu-id="72d9b-129">参照型、およびプロセスの要素の 1 つをスローしようを格納する配列の要素を列挙している、<xref:System.NullReferenceException>例外。</span><span class="sxs-lookup"><span data-stu-id="72d9b-129">You're enumerating the elements of an array that contains reference types, and your attempt to process one of the elements throws a <xref:System.NullReferenceException> exception.</span></span>  
  
     <span data-ttu-id="72d9b-130">次の例では、文字列の配列を定義します。</span><span class="sxs-lookup"><span data-stu-id="72d9b-130">The following example defines a string array.</span></span> <span data-ttu-id="72d9b-131">A`for`ステートメントは、配列内の要素を列挙し、呼び出す各文字列の<xref:System.String.Trim%2A>文字列を表示する前にメソッドです。</span><span class="sxs-lookup"><span data-stu-id="72d9b-131">A `for` statement enumerates the elements in the array and calls each string's <xref:System.String.Trim%2A> method before displaying the string.</span></span>  
  
     [!code-csharp[System.NullReferenceException.Class#8](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.nullreferenceexception.class/cs/Array1.cs#8)]
     [!code-vb[System.NullReferenceException.Class#8](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.nullreferenceexception.class/vb/Array1.vb#8)]  
  
     <span data-ttu-id="72d9b-132">配列の各要素が null 以外の値を含める必要があります、配列要素の値が実際と仮定した場合、この例外が発生した`null`です。</span><span class="sxs-lookup"><span data-stu-id="72d9b-132">This exception occurs if you assume that each element of the array must contain a non-null value, and the value of the array element is in fact `null`.</span></span> <span data-ttu-id="72d9b-133">要素があるかどうかをテストして、例外を排除できる`null`その要素に任意の操作を実行する前に次の例として示します。</span><span class="sxs-lookup"><span data-stu-id="72d9b-133">The exception can be eliminated by testing whether the element is `null` before performing any operation on that element, as the following example shows.</span></span>  
  
     [!code-csharp[System.NullReferenceException.Class#9](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.nullreferenceexception.class/cs/Array2.cs#9)]
     [!code-vb[System.NullReferenceException.Class#9](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.nullreferenceexception.class/vb/Array2.vb#9)]  
  
-   <span data-ttu-id="72d9b-134">A<xref:System.NullReferenceException>に渡されるメソッドが例外をスロー`null`です。</span><span class="sxs-lookup"><span data-stu-id="72d9b-134">A <xref:System.NullReferenceException> exception is thrown by a method that is passed `null`.</span></span> <span data-ttu-id="72d9b-135">一部のメソッドに渡される引数を検証します。</span><span class="sxs-lookup"><span data-stu-id="72d9b-135">Some methods validate the arguments that are passed to them.</span></span> <span data-ttu-id="72d9b-136">ように設定すると、引数の 1 つは`null`がスローされます、<xref:System.ArgumentNullException?displayProperty=nameWithType>例外。</span><span class="sxs-lookup"><span data-stu-id="72d9b-136">If they do and one of the arguments is `null`, the method throws an <xref:System.ArgumentNullException?displayProperty=nameWithType> exception.</span></span> <span data-ttu-id="72d9b-137">それ以外の場合、スロー、<xref:System.NullReferenceException>例外。</span><span class="sxs-lookup"><span data-stu-id="72d9b-137">Otherwise, it throws a <xref:System.NullReferenceException> exception.</span></span> <span data-ttu-id="72d9b-138">次の例では、このシナリオを示します。</span><span class="sxs-lookup"><span data-stu-id="72d9b-138">The following example illustrates this scenario.</span></span>  
  
     [!code-csharp[System.NullReferenceException.Class#3](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.nullreferenceexception.class/cs/example2.cs#3)]
     [!code-vb[System.NullReferenceException.Class#3](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.nullreferenceexception.class/vb/example2.vb#3)]  
  
     <span data-ttu-id="72d9b-139">この問題に対処する必ずメソッドに渡される引数がない`null`でスローされた例外を処理または、`try…catch…finally`ブロックします。</span><span class="sxs-lookup"><span data-stu-id="72d9b-139">To address this issue, make sure that the argument passed to the method is not `null`, or handle the thrown exception in a `try…catch…finally` block.</span></span> <span data-ttu-id="72d9b-140">詳細については、次を参照してください。[例外](http://msdn.microsoft.com/library/f99a1d29-a2a8-47af-9707-9909f9010735)です。</span><span class="sxs-lookup"><span data-stu-id="72d9b-140">For more information, see [Exceptions](http://msdn.microsoft.com/library/f99a1d29-a2a8-47af-9707-9909f9010735).</span></span>  
  
 <span data-ttu-id="72d9b-141">次の Microsoft intermediate language (MSIL) の命令をスロー <xref:System.NullReferenceException>: `callvirt`、 `cpblk`、 `cpobj`、 `initblk`、 `ldelem.<type>`、 `ldelema`、 `ldfld`、 `ldflda`、 `ldind.<type>`、 `ldlen`、 `stelem.<type>`、 `stfld`、 `stind.<type>`、 `throw`、および`unbox`です。</span><span class="sxs-lookup"><span data-stu-id="72d9b-141">The following Microsoft intermediate language (MSIL) instructions throw <xref:System.NullReferenceException>: `callvirt`, `cpblk`, `cpobj`, `initblk`, `ldelem.<type>`, `ldelema`, `ldfld`, `ldflda`, `ldind.<type>`, `ldlen`, `stelem.<type>`, `stfld`, `stind.<type>`, `throw`, and `unbox`.</span></span>  
  
 <span data-ttu-id="72d9b-142"><xref:System.NullReferenceException> 値は 0x80004003 HRESULT COR_E_NULLREFERENCE を使用します。</span><span class="sxs-lookup"><span data-stu-id="72d9b-142"><xref:System.NullReferenceException> uses the HRESULT COR_E_NULLREFERENCE, which has the value 0x80004003.</span></span>  
  
 <span data-ttu-id="72d9b-143">インスタンスの初期プロパティ値の一覧については<xref:System.NullReferenceException>を参照してください、<xref:System.NullReferenceException.%23ctor%2A>コンス トラクターです。</span><span class="sxs-lookup"><span data-stu-id="72d9b-143">For a list of initial property values for an instance of <xref:System.NullReferenceException>, see the <xref:System.NullReferenceException.%23ctor%2A> constructors.</span></span>  
  
 <span data-ttu-id="72d9b-144">**リリース コードにおける NullReferenceException の処理**</span><span class="sxs-lookup"><span data-stu-id="72d9b-144">**Handling NullReferenceException in release code**</span></span>  
  
 <span data-ttu-id="72d9b-145">通常発生後に処理するよりも、NullReferenceException を回避することをお勧めします。</span><span class="sxs-lookup"><span data-stu-id="72d9b-145">It's usually better to avoid a NullReferenceException than to handle it after it occurs.</span></span> <span data-ttu-id="72d9b-146">例外の処理によって、コードは管理や理解が難しくなることがあるうえ、他のバグが生じることも考えられます。</span><span class="sxs-lookup"><span data-stu-id="72d9b-146">Handling an exception can make your code harder to maintain and understand, and can sometimes introduce other bugs.</span></span> <span data-ttu-id="72d9b-147">NullReferenceException は多くの場合、回復不能なエラーです。</span><span class="sxs-lookup"><span data-stu-id="72d9b-147">A NullReferenceException is often a non-recoverable error.</span></span> <span data-ttu-id="72d9b-148">こうした場合、例外によってアプリを停止させることが最良の代替策である可能性があります。</span><span class="sxs-lookup"><span data-stu-id="72d9b-148">In these cases, letting the exception stop the app might be the best alternative.</span></span>  
  
 <span data-ttu-id="72d9b-149">ただし、次のように、エラーの処理が有用である状況は多く存在します。</span><span class="sxs-lookup"><span data-stu-id="72d9b-149">However, there are many situations where handling the error can be useful:</span></span>  
  
-   <span data-ttu-id="72d9b-150">アプリは null であるオブジェクトを無視できます。</span><span class="sxs-lookup"><span data-stu-id="72d9b-150">Your app can ignore objects that are null.</span></span> <span data-ttu-id="72d9b-151">たとえば、アプリがデータベース内のレコードを取得して処理する場合、結果として null オブジェクトである、いくつかの不良レコードを無視できる可能性があります。</span><span class="sxs-lookup"><span data-stu-id="72d9b-151">For example, if your app retrieves and processes records in a database, you might be able to ignore some number of bad records that result in null objects.</span></span> <span data-ttu-id="72d9b-152">必要な処理は、ログ ファイルまたはアプリケーションの UI に不良データを記録することだけであることが考えられます。</span><span class="sxs-lookup"><span data-stu-id="72d9b-152">Recording the bad data in a log file or in the application UI might be all you have to do.</span></span>  
  
-   <span data-ttu-id="72d9b-153">例外から回復することができます。</span><span class="sxs-lookup"><span data-stu-id="72d9b-153">You can recover from the exception.</span></span> <span data-ttu-id="72d9b-154">たとえば、参照型を返す Web サービスの呼び出しは、接続が失われた場合や接続がタイムアウトした場合には、null を返す可能性があります。接続を再確立し、呼び出しを再度試すことができます。</span><span class="sxs-lookup"><span data-stu-id="72d9b-154">For example, a call to a web service that returns a reference type might return null if the connection is lost or the connection times out. You can attempt to reestablish the connection and try the call again.</span></span>  
  
-   <span data-ttu-id="72d9b-155">アプリの状態を有効な状態に復元できます。</span><span class="sxs-lookup"><span data-stu-id="72d9b-155">You can restore the state of your app to a valid state.</span></span> <span data-ttu-id="72d9b-156">たとえば、NullReferenceException をスローするメソッドを呼び出す前に、情報をデータ ストアに保存する必要がある、複数手順のタスクを実行していることがあります。</span><span class="sxs-lookup"><span data-stu-id="72d9b-156">For example, you might be performing a multi-step task that requires you to save information to a data store before you call a method that throws a NullReferenceException.</span></span> <span data-ttu-id="72d9b-157">初期化されていないオブジェクトがデータ レコードを破損する可能性がある場合、アプリを終了する前に過去のデータを削除できます。</span><span class="sxs-lookup"><span data-stu-id="72d9b-157">If the uninitialized object would corrupt the data record, you can remove the previous data before you close the app.</span></span>  
  
-   <span data-ttu-id="72d9b-158">例外の報告が推奨されます。</span><span class="sxs-lookup"><span data-stu-id="72d9b-158">You want to report the exception.</span></span> <span data-ttu-id="72d9b-159">たとえば、エラーがアプリのユーザーによる間違いに起因する場合、メッセージを生成して、ユーザーが正しい情報を入力するのを支援できます。</span><span class="sxs-lookup"><span data-stu-id="72d9b-159">For example, if the error was caused by a mistake from the user of your app, you can generate a message to help him supply the correct information.</span></span> <span data-ttu-id="72d9b-160">問題の解決に役立てるために、エラーに関する情報をログに記録することもできます。</span><span class="sxs-lookup"><span data-stu-id="72d9b-160">You can also log information about the error to help you fix the problem.</span></span> <span data-ttu-id="72d9b-161">ASP.NET などのいくつかのフレームワークには、アプリのクラッシュを防止するために、すべてのエラーをキャプチャする上位レベルの例外ハンドラーが用意されています。このハンドラーを使用した場合、例外をログに記録することが、その例外が発生したことを知る唯一の方法になる可能性があります。</span><span class="sxs-lookup"><span data-stu-id="72d9b-161">Some frameworks, like ASP.NET, have a high-level exception handler that captures all errors to that the app never crashes; in that case, logging the exception might be the only way you can know that it occurs.</span></span>  
  
 ]]></format>
    </remarks>
    <altmember cref="T:System.Exception" />
  </Docs>
  <Members>
    <MemberGroup MemberName=".ctor">
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
        <AssemblyVersion>4.0.20.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Docs>
        <summary>
          <span data-ttu-id="72d9b-162">
            <see cref="T:System.NullReferenceException" /> クラスの新しいインスタンスを初期化します。</span>
          <span class="sxs-lookup">
            <span data-stu-id="72d9b-162">Initializes a new instance of the <see cref="T:System.NullReferenceException" /> class.</span>
          </span>
        </summary>
      </Docs>
    </MemberGroup>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public NullReferenceException ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor() cil managed" />
      <MemberSignature Language="DocId" Value="M:System.NullReferenceException.#ctor" />
      <MemberSignature Language="VB.NET" Value="Public Sub New ()" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; NullReferenceException();" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
        <AssemblyVersion>4.0.20.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters />
      <Docs>
        <summary>
          <span data-ttu-id="72d9b-163">
            <see cref="T:System.NullReferenceException" /> クラスの新しいインスタンスを初期化し、その <see cref="P:System.Exception.Message" /> プロパティの値として、"オブジェクトのインスタンスを指定する必要がある部分に null 値が指定されています" などのシステム提供のエラーを指定します。</span>
          <span class="sxs-lookup">
            <span data-stu-id="72d9b-163">Initializes a new instance of the <see cref="T:System.NullReferenceException" /> class, setting the <see cref="P:System.Exception.Message" /> property of the new instance to a system-supplied message that describes the error, such as "The value 'null' was found where an instance of an object was required."</span>
          </span>
          <span data-ttu-id="72d9b-164">このメッセージには、現在のシステムのカルチャが考慮されます。</span>
          <span class="sxs-lookup">
            <span data-stu-id="72d9b-164">This message takes into account the current system culture.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="72d9b-165"><xref:System.NullReferenceException> のインスタンスの初期プロパティ値を次の表に示します。</span><span class="sxs-lookup"><span data-stu-id="72d9b-165">The following table shows the initial property values for an instance of <xref:System.NullReferenceException>.</span></span>  
  
|<span data-ttu-id="72d9b-166">プロパティ</span><span class="sxs-lookup"><span data-stu-id="72d9b-166">Property</span></span>|<span data-ttu-id="72d9b-167">[値]</span><span class="sxs-lookup"><span data-stu-id="72d9b-167">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="72d9b-168">`null`。</span><span class="sxs-lookup"><span data-stu-id="72d9b-168">`null`.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="72d9b-169">ローカライズされたエラー メッセージ文字列。</span><span class="sxs-lookup"><span data-stu-id="72d9b-169">The localized error message string.</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public NullReferenceException (string message);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.NullReferenceException.#ctor(System.String)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; NullReferenceException(System::String ^ message);" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
        <AssemblyVersion>4.0.20.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
      </Parameters>
      <Docs>
        <param name="message">
          <span data-ttu-id="72d9b-170">エラーを説明する <see cref="T:System.String" />。</span>
          <span class="sxs-lookup">
            <span data-stu-id="72d9b-170">A <see cref="T:System.String" /> that describes the error.</span>
          </span>
          <span data-ttu-id="72d9b-171">
            <c>メッセージ</c>の内容は、ユーザーが理解できる内容にします。</span>
          <span class="sxs-lookup">
            <span data-stu-id="72d9b-171">The content of <c>message</c> is intended to be understood by humans.</span>
          </span>
          <span data-ttu-id="72d9b-172">このコンストラクターの呼び出し元は、この文字列が現在のシステムのカルチャに合わせてローカライズ済みであることを確認しておく必要があります。</span>
          <span class="sxs-lookup">
            <span data-stu-id="72d9b-172">The caller of this constructor is required to ensure that this string has been localized for the current system culture.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="72d9b-173">指定したエラー メッセージを使用して、<see cref="T:System.NullReferenceException" /> クラスの新しいインスタンスを初期化します。</span>
          <span class="sxs-lookup">
            <span data-stu-id="72d9b-173">Initializes a new instance of the <see cref="T:System.NullReferenceException" /> class with a specified error message.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="72d9b-174"><xref:System.NullReferenceException> のインスタンスの初期プロパティ値を次の表に示します。</span><span class="sxs-lookup"><span data-stu-id="72d9b-174">The following table shows the initial property values for an instance of <xref:System.NullReferenceException>.</span></span>  
  
|<span data-ttu-id="72d9b-175">プロパティ</span><span class="sxs-lookup"><span data-stu-id="72d9b-175">Property</span></span>|<span data-ttu-id="72d9b-176">[値]</span><span class="sxs-lookup"><span data-stu-id="72d9b-176">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="72d9b-177">`null`。</span><span class="sxs-lookup"><span data-stu-id="72d9b-177">`null`.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="72d9b-178">エラー メッセージ文字列。</span><span class="sxs-lookup"><span data-stu-id="72d9b-178">The error message string.</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="protected NullReferenceException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);" />
      <MemberSignature Language="ILAsm" Value=".method familyhidebysig specialname rtspecialname instance void .ctor(class System.Runtime.Serialization.SerializationInfo info, valuetype System.Runtime.Serialization.StreamingContext context) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.NullReferenceException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)" />
      <MemberSignature Language="VB.NET" Value="Protected Sub New (info As SerializationInfo, context As StreamingContext)" />
      <MemberSignature Language="C++ CLI" Value="protected:&#xA; NullReferenceException(System::Runtime::Serialization::SerializationInfo ^ info, System::Runtime::Serialization::StreamingContext context);" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="info" Type="System.Runtime.Serialization.SerializationInfo" />
        <Parameter Name="context" Type="System.Runtime.Serialization.StreamingContext" />
      </Parameters>
      <Docs>
        <param name="info">
          <span data-ttu-id="72d9b-179">シリアル化されたオブジェクト データを保持するオブジェクト。</span>
          <span class="sxs-lookup">
            <span data-stu-id="72d9b-179">The object that holds the serialized object data.</span>
          </span>
        </param>
        <param name="context">
          <span data-ttu-id="72d9b-180">転送元または転送先に関するコンテキスト情報。</span>
          <span class="sxs-lookup">
            <span data-stu-id="72d9b-180">The contextual information about the source or destination.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="72d9b-181">シリアル化したデータを使用して、<see cref="T:System.NullReferenceException" /> クラスの新しいインスタンスを初期化します。</span>
          <span class="sxs-lookup">
            <span data-stu-id="72d9b-181">Initializes a new instance of the <see cref="T:System.NullReferenceException" /> class with serialized data.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="72d9b-182">このコンストラクターは、逆シリアル化中に、ストリームで送信された例外オブジェクトを再構築するために呼び出されます。</span><span class="sxs-lookup"><span data-stu-id="72d9b-182">This constructor is called during deserialization to reconstitute the exception object transmitted over a stream.</span></span> <span data-ttu-id="72d9b-183">詳細については、次を参照してください。 [XML および SOAP シリアル化](~/docs/standard/serialization/xml-and-soap-serialization.md)です。</span><span class="sxs-lookup"><span data-stu-id="72d9b-183">For more information, see [XML and SOAP Serialization](~/docs/standard/serialization/xml-and-soap-serialization.md).</span></span>  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public NullReferenceException (string message, Exception innerException);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message, class System.Exception innerException) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.NullReferenceException.#ctor(System.String,System.Exception)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String, innerException As Exception)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; NullReferenceException(System::String ^ message, Exception ^ innerException);" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
        <AssemblyVersion>4.0.20.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
        <Parameter Name="innerException" Type="System.Exception" />
      </Parameters>
      <Docs>
        <param name="message">
          <span data-ttu-id="72d9b-184">例外の原因を説明するエラー メッセージ。</span>
          <span class="sxs-lookup">
            <span data-stu-id="72d9b-184">The error message that explains the reason for the exception.</span>
          </span>
        </param>
        <param name="innerException">
          <span data-ttu-id="72d9b-185">現在の例外の原因となった例外。</span>
          <span class="sxs-lookup">
            <span data-stu-id="72d9b-185">The exception that is the cause of the current exception.</span>
          </span>
          <span data-ttu-id="72d9b-186">
            <c>innerException</c> パラメーターが <see langword="null" /> でない場合は、内部例外を処理する <see langword="catch" /> ブロックで現在の例外が発生します。</span>
          <span class="sxs-lookup">
            <span data-stu-id="72d9b-186">If the <c>innerException</c> parameter is not <see langword="null" />, the current exception is raised in a <see langword="catch" /> block that handles the inner exception.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="72d9b-187">指定したエラー メッセージおよびこの例外の原因となった内部例外への参照を使用して、<see cref="T:System.NullReferenceException" /> クラスの新しいインスタンスを初期化します。</span>
          <span class="sxs-lookup">
            <span data-stu-id="72d9b-187">Initializes a new instance of the <see cref="T:System.NullReferenceException" /> class with a specified error message and a reference to the inner exception that is the cause of this exception.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="72d9b-188">前の例外の直接の結果としてスローされる例外については、<xref:System.Exception.InnerException%2A> プロパティに、前の例外への参照が格納されます。</span><span class="sxs-lookup"><span data-stu-id="72d9b-188">An exception that is thrown as a direct result of a previous exception should include a reference to the previous exception in the <xref:System.Exception.InnerException%2A> property.</span></span> <span data-ttu-id="72d9b-189"><xref:System.Exception.InnerException%2A> プロパティは、コンストラクターに渡されたものと同じ値を返します。`null` プロパティによって内部例外値がコンストラクターに渡されなかった場合は、<xref:System.Exception.InnerException%2A> を返します。</span><span class="sxs-lookup"><span data-stu-id="72d9b-189">The <xref:System.Exception.InnerException%2A> property returns the same value that is passed into the constructor, or `null` if the <xref:System.Exception.InnerException%2A> property does not supply the inner exception value to the constructor.</span></span>  
  
 <span data-ttu-id="72d9b-190"><xref:System.NullReferenceException> のインスタンスの初期プロパティ値を次の表に示します。</span><span class="sxs-lookup"><span data-stu-id="72d9b-190">The following table shows the initial property values for an instance of <xref:System.NullReferenceException>.</span></span>  
  
|<span data-ttu-id="72d9b-191">プロパティ</span><span class="sxs-lookup"><span data-stu-id="72d9b-191">Property</span></span>|<span data-ttu-id="72d9b-192">[値]</span><span class="sxs-lookup"><span data-stu-id="72d9b-192">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="72d9b-193">内部例外の参照。</span><span class="sxs-lookup"><span data-stu-id="72d9b-193">The inner exception reference.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="72d9b-194">エラー メッセージ文字列。</span><span class="sxs-lookup"><span data-stu-id="72d9b-194">The error message string.</span></span>|  
  
 ]]></format>
        </remarks>
        <altmember cref="T:System.Exception" />
      </Docs>
    </Member>
  </Members>
</Type>